src/ser/mod.rs

```diff
@@ -489,6 +522,12 @@ impl<W: fmt::Write> Serializer<W> {
             .map_or(true, |(ref config, _)| config.escape_strings)
     }
 
+    fn hex_as_raw(&self) -> bool {
+        self.pretty
+            .as_ref()
+            .map_or(true, |(ref config, _)| config.hex_as_raw)
```

Comment:
This implies that the `hex_as_raw` option is enabled when no pretty config is given. Same with the default in the `PrettyConfig` above. Usually pretty config options are opt-in, so I'd like to hear some justification for being opt-out.